File: integrations/sdk/python/plugin_client.py
"""Thin Plugin Runtime API client (spec 014)."""

from __future__ import annotations

import json
import urllib.request
from typing import Any


class PluginClient:
    def __init__(self, base_url: str) -> None:
        self._base_url = base_url.rstrip("/")

    def handle(self, event: dict[str, Any]) -> dict[str, Any]:
        body = json.dumps(event).encode("utf-8")
        req = urllib.request.Request(
            f"{self._base_url}/v1/plugin/handle",
            data=body,
            headers={"Content-Type": "application/json"},
            method="POST",
        )
        with urllib.request.urlopen(req, timeout=30) as resp:
            return json.loads(resp.read().decode("utf-8"))

File: integrations/sdk/php/PluginClient.php
<?php
declare(strict_types=1);

namespace Korus\PluginSdk;

final class PluginClient
{
    public function __construct(private readonly string $baseUrl) {}

    /** @param array<string, mixed> $event */
    public function handle(array $event): array
    {
        $json = json_encode($event, JSON_THROW_ON_ERROR);
        $ctx = stream_context_create([
            'http' => [
                'method' => 'POST',
                'header' => "Content-Type: application/json\r\n",
                'content' => $json,
                'timeout' => 30,
            ],
        ]);
        $url = rtrim($this->baseUrl, '/') . '/v1/plugin/handle';
        $raw = file_get_contents($url, false, $ctx);
        if ($raw === false) {
            throw new \RuntimeException('plugin request failed');
        }
        /** @var array<string, mixed> */
        return json_decode($raw, true, 512, JSON_THROW_ON_ERROR);
    }
}
